rainbows.rb

```diff
@@ -1,29 +1,52 @@
-red = [255,0,0]
-green = [0,255,0]
-blue = [0,0,255]
-
-color_lookup = {
-  [255,0,0] => "red",
-  [255,125,0] => "orange",
-  [255,255,0] => "yellow",
-  [125,255,0] => "spring green",
-  [0,255,0] => "green",
-  [0,255,125] => "turquoise",
-  [0,255,255] => "cyan",
-  [0,125,255] => "ocean",
-  [0,0,255] => "blue",
-  [125,0,255] => "violet",
-  [255,0,255] => "magenta",
-  [255,0,125] => "raspberry"
-}
-
-def mix_colors(color1,color2)
-  mixed_color = []
-  3.times do |i|
-    mixed_color[i] = color1[i] + color2[i]
+require 'colorize'
+
+def get_color (red, green, blue)
+
+  rainbow = [
+    { red: 
+      { r: 255, g: 0, b: 0 } 
+    },
+
+    { orange: 
+      { r: 255, g: 127, b: 0 } 
+    },
+
+    { yellow: 
+      { r: 255, g: 255, b: 0 }
+    },
+
+    { green: 
+      { r: 0, g: 255, b: 0 }
+    },
+
+    { blue: 
+      { r: 0, g: 0, b: 255 }
+    },
+
+    { indigo: 
+      { r: 75, g: 0, b: 130 }
+    },
+    
+    { violet:
+     { r: 143, g: 0, b: 255 }
+    },
+  ]
+
+  rainbow.each do |color|
+    color.each do |key, value|
+      if red == value[:r] && green == value[:g] && blue == value[:b]
+        return key.to_s
+      else
+        puts "Sorry, I do not know that color!"
+      end
+    end
```

Comment:
This is good, but ignores the ordering of the array and makes for a less efficient solution. It works, which is great! But there are ways to optimize the solution. When we go over in class, bring this up, if I forget.